CONTRIBUTING.md

```diff
@@ -0,0 +1,192 @@
+
+**Please ask first** before embarking on any significant pull request (e.g. implementing features, refactoring code), otherwise you risk spending a lot of time working on something that the project's developers might not want to merge into the project.
+
+### Setting up the project in your local machine
+
+1. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the project, clone your fork, and configure the remotes:
+
+   ```bash
+   # Clone your fork of the repo into the current directory
+   git clone https://github.com/<your-username>/<repo-name>
+
+   # Navigate to the newly cloned directory
+   cd <repo-name>
+
+   # Assign the original repo to a remote called "upstream"
+   git remote add upstream https://github.com/openpgpjs/<repo-name>
+   ```
+
+2. Install dependencies with your preferred package manager:
+
+   ```bash
+   npm install
+   ```
+
+3. Create a new topic branch (off the main project development branch) to contain your feature, change, or fix:
+
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+4. Follow our [coding conventions](#coding-conventions) while developing your changes. You can also run `npm run lint` to match our requirements.
+
+5. Write clear and meaningful git commit messages. Please follow our [commit message conventions](#commit-conventions) while committing to your branch.
+
+6. Make sure to update or add to the tests when appropriate. Run the appropriate testing suites to check that all tests pass after you've made changes. You can read about our different types of tests in our [Testing](#testing) section.
+
+7. If you added or changed a feature, make sure to document it accordingly in the [README.md](https://github.com/openpgpjs/openpgpjs/blob/main/README.md) file.
+
+## Coding conventions
+
+We ensure code consistency through our [ESLint config](https://github.com/openpgpjs/openpgpjs/blob/main/.eslintrc.js). You can run `npm run lint` to lint your changes. When in doubt, follow the style of the surrounding code.
+
+## Commit conventions
+
+We follow the seven rules of a great Git commit message, which is extensively described in [this blog post](https://cbea.ms/git-commit/), but in short the principles are:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+
+You can also take a look at our [main branch commits](https://github.com/openpgpjs/openpgpjs/commits/main).
+
+## Testing
+
+We have multiple types of tests. They are located in the root [`/test`](https://github.com/openpgpjs/openpgpjs/tree/main/test) folder.
+Before you can run the tests you need to build the project using `npm run build` and this will bundle the project and the tests for it be used in our different testing environments.
```

Comment:
To simplify things, IMO it's fine to leave the redundant suggestion to re-run `npm run build`. Then also add `npm run build-test`. That would should all cases?

Reply:
`npm run build-test` is part of `npm run browsertest`, so shouldn't be necessary. `npm run build` can be useful to run after making changes (and before running the tests), but then I'd rather say that, e.g.:

```suggestion
Before running the tests, run `npm run build` to build OpenPGP.js if you've made changes since running `npm install`.
```

Reply:
> `npm run build-test` is part of `npm run browsertest`

but it's not part of `npm run test-browser` and `-browserstack`. I just think it's less confusing to present a consistent procedure; it doesn't have side-effects anyway.